pkg/gce-pd-csi-driver/controller.go

```diff
@@ -391,22 +393,28 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 		}
 		disk, err = createSingleZoneDisk(ctx, gceCS.CloudProvider, name, zones, params, capacityRange, capBytes, snapshotID, volumeContentSourceVolumeID, multiWriter)
 		if err != nil {
-			return nil, LoggedError("CreateVolume failed to create single zonal disk "+name+": ", err)
+			// Emit metric for expected disk type from storage class
+			defer gceCS.Metrics.RecordOperationErrorMetrics("CreateVolume", err, params.DiskType)
```

Comment:
What I mean with using defer, is to add the defer statement at the top of the function. That way every return branch will emit a metric, and if we add more return branches in the future, we won't forget to emit a metric for it. So something like

```
func CreateVolume() {
  // declare some variables used by the metrics recording
  err error
  diskType string

  defer func() {
    if err != nil {
      recordErrorMetric("CreateVolume", err, diskType)
    }
  }{}

  // rest of the function
}
```